test/cxx/Core/ApplicationPool/PoolTest.cpp

```diff
@@ -190,7 +191,16 @@ namespace tut {
 				P_ERROR("get() exception: " << currentException->what());
 				abort();
 			}
-			currentSession->initiate();
+			if (!currentSession->initiate()) {
+				unsigned long long timeout = 10000;
+				while (timeout > 0 && !waitUntilWritable(currentSession->fd(), &timeout)) { }
```

Comment:
A while loop is not appropriate here. waitUntilWritable works just like poll. If it returns false then that just means the full timeout expired, and you should fail the test.

Reply:
It looks like it breaks the timeout into 1000 chunks and adjusts the timeout pointer as it is called repeatedly… https://github.com/phusion/passenger/blob/stable-6.0/src/cxx_supportlib/IOTools/IOUtils.cpp#L816 https://github.com/phusion/passenger/blob/stable-6.0/src/cxx_supportlib/IOTools/IOUtils.cpp#L864-L879

Reply:
The timeout is in microseconds, that's why the /1000.